File: contests/ABC/ABC429/C/solution.rb
N = gets.to_i
A = gets.split.map(&:to_i)

hash = Hash.new()
A.each do |val|
  if hash.has_key?(val)
    hash[val] += 1
  else
    hash[val] = 1
  end
end

res = 0
(hash.values).each do |cnt|
  res += (cnt*(cnt-1)/2) * (N-cnt)
end
puts res

File: contests/ABC/ABC429/C/solution.py
N = int(input())
A = list(map(int, input().split()))

dict = {}
for i,v in enumerate(A):
    if not v in dict:
        dict[v] = 1
    else:
        dict[v] += 1

res = 0
for v in dict.values():
    if v >= 2:
        res += (v * (v - 1) // 2) * (N - v)
        

print(res)